packages/at_secondary_server/lib/src/connection/outbound/outbound_client.dart

```diff
@@ -67,8 +68,25 @@ class OutboundClient {
   /// Throws a [SecondaryNotFoundException] if secondary is url is not found for atsign
   /// Throws a [SocketException] when a socket connection to secondary cannot be established
   /// Throws a [HandShakeException] for any exception in the handshake process
-  Future<bool> connect({bool handshake = true}) async {
-    logger.finer('connect(handshake:$handshake) called for $toAtSign');
+  Future<bool> connect({required bool handshake}) async {
+    // TODO There is the potential for races here under heavy load.
+    // TODO Need to make isHandshakeDone and isConnectionCreated into futures
+    // which return the futures of underlying completers which are completed
+    // when the appropriate thing has happened.
+    if (handshake != handshakeRequired) {
```

Comment:
I have reviewed the full codebase and this is never the case right now. This check has been added for the safety of future maintainers.